freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java

```diff
@@ -47,48 +47,58 @@
-        if (catalogUrl == null) {
-            LOGGER.info("No image catalog was defined!");
-            return null;
-        }
-
         try {
-            long started = System.currentTimeMillis();
-            String content;
-            if (catalogUrl.startsWith("http")) {
-                Client client = RestClientUtil.get();
-                WebTarget target = client.target(catalogUrl);
-                Response response = target.request().get();
-                content = readResponse(target, response);
-            } else {
-                content = readCatalogFromFile(catalogUrl);
+            if (Objects.nonNull(catalogUrl)) {
+                long started = System.currentTimeMillis();
+                String content = readCatalogContent(catalogUrl);
+                ImageCatalog catalog = objectMapper.readValue(content, ImageCatalog.class);
+                if (Objects.nonNull(catalog)) {
+                    ImageCatalog filteredCatalog = filterImagesByOsType(catalog);
+                    long timeOfParse = System.currentTimeMillis() - started;
+                    LOGGER.debug("ImageCatalog was fetched and parsed from '{}' and took '{}' ms.", catalogUrl, timeOfParse);
+                    return filteredCatalog;
+                }
+                throw new ImageCatalogException(String.format("Failed to read an image catalog from '%s' content.", content));
             }
-            catalog = filterImagesByOsType(objectMapper.readValue(content, ImageCatalog.class));
-            long timeOfParse = System.currentTimeMillis() - started;
-            LOGGER.debug("ImageCatalog was fetched and parsed from '{}' and took '{}' ms.", catalogUrl, timeOfParse);
+            throw new ImageCatalogException("Unable to fetch image catalog by the catalogUrl is null.");
+        } catch (ImageCatalogException e) {
+            throw e;
         } catch (RuntimeException e) {
             throw new ImageCatalogException(String.format("Failed to get image catalog: %s from %s", e.getCause(), catalogUrl), e);
         } catch (JsonMappingException e) {
             throw new ImageCatalogException(String.format("Invalid json format for image catalog with error: %s", e.getMessage()), e);
         } catch (IOException e) {
             throw new ImageCatalogException(String.format("Failed to read image catalog from file: '%s'", catalogUrl), e);
         }
-        return catalog;
     }
 
     @CacheEvict(value = "imageCatalogCache", key = "#catalogUrl")
     public void evictImageCatalogCache(String catalogUrl) {
     }
 
+    private String readCatalogContent(String catalogUrl) throws IOException {
+        if (catalogUrl.startsWith("http")) {
+            Client client = RestClientUtil.get();
+            WebTarget target = client.target(catalogUrl);
+            Response response = target.request().get();
+            return readResponse(target, response);
+        } else {
+            return readCatalogFromFile(catalogUrl);
```

Comment:
As I see it reads from `/etc/freeipa` folder and if the content is not a valid json it throws the content in the error message. Is it possible anyhow that the user can get secrets this way? I didn't really try it out but I don't see any obstacle to do it.

Reply:
I really like this raise of possible security leak. Thank you. I created a unit test to cover it.

Reply:
Cool thanks!